modules/nextflow/src/main/groovy/nextflow/script/ScriptMeta.groovy

```diff
@@ -170,14 +170,15 @@ class ScriptMeta {
 
     static List<FunctionDef> definedFunctions0(BaseScript script) {
         final allMethods = script.class.getDeclaredMethods()
-        final result = new ArrayList(allMethods.length)
+        final result = new ArrayList<FunctionDef>(allMethods.length)
         for( Method method : allMethods ) {
             if( !Modifier.isPublic(method.getModifiers()) ) continue
             if( Modifier.isStatic(method.getModifiers())) continue
             if( method.name.startsWith('super$')) continue
             if( method.name in INVALID_FUNCTION_NAMES ) continue
 
-            result.add(new FunctionDef(script, method))
+            if( result.find{it.name == method.name}) continue
```

Comment:
What this does?

Reply:
simply avoid to have duplicated FunctionDef objects in the list